server/@types/viewModels/index.d.ts

```diff
@@ -107,7 +107,7 @@ declare module 'viewModels' {
     courseName: string
     courseCode: string
     courseStartDate: Date
-    courseCompleted: boolean
+    courseStatus: 'COMPLETED' | 'IN_PROGRESS' | 'WITHDRAWN' | 'TEMPORARILY_WITHDRAWN'
```

Comment:
Update our view model by removing the `courseCompleted: boolean` field, and replacing it with a `courseStatus` field of an enum('ish) type.
Essentially this is the main ask of this PR - we need our view model representation of a Curious course to have a definitive status so that we can group and display them onscreen.

Everything else in this PR is to support this view model change, be it changes to the mapper, tests, test data builders etc